### PR TITLE
Bump alpine to 3.19

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
Turns out the cloud-init-local service is now vital for cloud-init to work on Alpine:
https://leo.leung.xyz/wiki/Cloud-init
https://gitlab.com/redhat/centos-stream/rpms/cloud-init/-/blob/020655f489ca26b5830de8423354a14ccd0552db/cloud-init.spec#L138
I've double checked cloud-init breaks when I remove it.
(`quay.io/akalenyu/alpine-cloud-init:3.19-broken` vs `quay.io/akalenyu/alpine-cloud-init:3.19`)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by andreabolognani** [comment]
@akalenyu thank you again for your amazing work here! You really got us out of the rut we were stuck in.

I've tested the updated image and the NoCloud volume is finally being taken into consideration. However, we also need

```diff
diff --git a/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh b/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh
index f88e0265..4e3b7656 100755
--- a/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh
+++ b/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh
@@ -25,4 +25,5 @@ rc-update add cloud-init default
 rc-update add cloud-init-local default
 rc-update add cloud-config default
 rc-update add cloud-final default
+rc-update add networking default
 rc-update add sshd default
```

otherwise the network interfaces will not be brought up.

File: cluster-provision/images/vm-image-builder/alpine-cloud-init/create-image.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ docker run --rm --platform=$PLATFORM -v /lib/modules:/lib/modules -v /dev:/dev -
 ./alpine-make-vm-image \
 	--image-format qcow2 \
 	--image-size 200M \
-    --branch v3.16 \
```
Comment (andreabolognani) [inline cluster-provision/images/vm-image-builder/alpine-cloud-init/create-image.sh:27]: Can you please fix the alignment while you're at it? It really bugs me that this line doesn't sit flush with the surrounding ones O:-)

Comment (akalenyu) [reply cluster-provision/images/vm-image-builder/alpine-cloud-init/create-image.sh:27]: sure, done

**Comment by akalenyu** [comment]
> @akalenyu thank you again for your amazing work here! You really got us out of the rut we were stuck in.
> 
> I've tested the updated image and the NoCloud volume is finally being taken into consideration. However, we also need
> 
> ```diff
> diff --git a/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh b/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh
> index f88e0265..4e3b7656 100755
> --- a/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh
> +++ b/cluster-provision/images/vm-image-builder/alpine-cloud-init/configure.sh
> @@ -25,4 +25,5 @@ rc-update add cloud-init default
>  rc-update add cloud-init-local default
>  rc-update add cloud-config default
>  rc-update add cloud-final default
> +rc-update add networking default
>  rc-update add sshd default
> ```
> 
> otherwise the network interfaces will not be brought up.

thanks, done
wonder what changed now, that made us do it explicitly now

**Comment by andreabolognani** [comment]
Can this be moved forward? My PR that needs it (https://github.com/kubevirt/kubevirt/pull/11256) is waiting for some changes on the CentOS Stream side, but we could do the bump to Alpine 3.19 in KubeVirt separately in the meantime, sort of get ahead of the game in a sense...

**Comment by akalenyu** [comment]
> Can this be moved forward? My PR that needs it ([kubevirt/kubevirt#11256](https://github.com/kubevirt/kubevirt/pull/11256)) is waiting for some changes on the CentOS Stream side, but we could do the bump to Alpine 3.19 in KubeVirt separately in the meantime, sort of get ahead of the game in a sense...

/cc @dhiller

**Review by dhiller — APPROVED** [review]
/approve

Thanks @akalenyu !

**Comment by dhiller** [comment]
^^ looked like a hiccup

**Comment by andreabolognani** [comment]
@akalenyu @dhiller I don't see an updated image at https://quay.io/repository/kubevirtci/alpine-with-test-tooling-container-disk?tab=tags, does a job need to be kicked off manually or something?

**Comment by akalenyu** [comment]
> @akalenyu @dhiller I don't see an updated image at https://quay.io/repository/kubevirtci/alpine-with-test-tooling-container-disk?tab=tags, does a job need to be kicked off manually or something?

There should be a tag containing this commit in
https://github.com/kubevirt/kubevirtci/tags
(which would also handle publishing the alpine img, IIUC)
But for some reason quite some time passed and this didn't happen

**Comment by andreabolognani** [comment]
Still no image AFAICT...

**Comment by dhiller** [comment]
> Still no image AFAICT...

Hey @akalenyu @andreabolognani image should be there now - latest image tag is `2404181910-1c58677`

**Comment by andreabolognani** [comment]
@dhiller yup, it's there now. Thanks a lot! For future reference, how did you make it happen? Did some automation need to be fixed, some job need to be kicked off manually, something else...

**Pull request merged**